src/Controller/CustomNodeEditController.php

```diff
@@ -50,60 +43,50 @@ public static function create(ContainerInterface $container) {
   }
 
   /**
-   * Returns form modes active for current user
-   *
-   * @return array 
+   * Returns form modes active for current user.
    *
+   * @return array
+   *   The list of active modes for the current user.
    */
-  public function getActiveNodeFormModes() {
-    $activeFormModes = [];
-    // "default" form mode is not returned from getFormModes(). Only named form modes.
-    $formModes = $this->entityDisplayRepository->getFormModes('node');
-    // Check which form modes user has access to
-    foreach ($formModes as $key => $value) {
-      if ($this->currentUser()->hasPermission("use {$value['id']} form mode")) {
-        $activeFormModes[$key] = $value;
-      }
-    }
-    
-    return $activeFormModes;
+  public function getActiveNodeFormModes(NodeInterface $node) {
+    $formModes = $this->entityDisplayRepository->getFormModeOptionsByBundle('node', $node->bundle());
+    return array_filter($formModes, function ($mode) {
+      return $this->currentUser()->hasPermission("use node.$mode form mode");
```

Comment:
Nice. Let's use {$mode} to be consistent with {$firstActiveFormMode}. It is just cosmetics I know. This is a side note, I will in general (and I will make exceptions) for expected large Arrays prefer an foreach instead of an array filter (for performance) but in this case filter reads better and the array is always expected to be small, so good, good!

Reply:
Oh I am a big fan of closures :)